pkg/storage/ec/client.go

```diff
@@ -66,7 +67,8 @@ func (ec *ecClient) Put(ctx context.Context, nodes []*proto.Node, rs eestream.Re
 	if !unique(nodes) {
 		return Error.New("duplicated nodes are not allowed")
 	}
-	readers, err := eestream.EncodeReader(ctx, data, rs, ec.mbm)
+	padded := eestream.PadReader(ioutil.NopCloser(data), rs.DecodedBlockSize())
```

Comment:
wanna put a todo that we should take this out and make sure padding is happening an a higher level with encryption? or should we just always have unencrypted padding?

Reply:
This is why I initially left the padding out of the ECClient. I thought another layer will take care of it. But we still don't have such layer. And I am not really sure it's a responsibility of the Segment Store to do the padding, which is something specific to erasure coding.

Thus, I the moment I think the ECClient should always make sure that the data is padded.

Reply:
well really it's whereever encryption happens, since that also requires padding. maybe we put it in the passthrough streamstore for now?

Reply:
What if there is no encryption involved? I believe we will have such use cases.

If we put this in the stream store then the data that is stored as inline segments will also be padded. Every byte of metadata counts, right?

Wherever we make the encryption, we might have another padding. In such case the data will come padded to ECClient and then the padding in ECClient will be no-op.